test/e2e/apps/cronjob.go

```diff
@@ -261,7 +261,7 @@ var _ = SIGDescribe("CronJob", func() {
 
 		ginkgo.By("Ensuring job was deleted")
 		_, err = e2ejob.GetJob(ctx, f.ClientSet, f.Namespace.Name, job.Name)
-		framework.ExpectError(err)
+		gomega.Expect(err).To(gomega.HaveOccurred())
```

Comment:
You can combine this with the next line. Basically what the test expects here is a `NotFound` error, let's find a way to express that.

The `Failed to delete %s cronjob in namespace %s"` failure message is bad: it gets emitted when a) some error occurred (previous check) and b) that error isn't the "NotFound" error. Instead, it should have been something like `"Unexpected error occurred: %v", err`, which is exactly what `MatchError` will produce.

Reply:
Patrick, thanks your comments and detail explanation.  will update  as soon as possible.